### PR TITLE
chore: apply and fix flake8-bugbear lint rules

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

- apply `flake-bugbear` rules to prevent and reduce likely bugs and usages
  - ruff rules list: https://docs.astral.sh/ruff/rules/#flake8-bugbear-b
  - PyCQA/flake8-bugbear: https://github.com/PyCQA/flake8-bugbear
- fixed the violations in the following list:
  - B033 duplicate value in set values in `Python3TemplateTransformer`
  - B024 marked abstract base class but no abstract methods in the class `Callback`
  - B018 touching the existed fields but never assigned or used in `ToolManager` and `ToolNode`

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
CC @takatost @crazywoola

File: api/core/helper/code_executor/python3/python3_transformer.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 PYTHON_STANDARD_PACKAGES = {
     'json', 'datetime', 'math', 'random', 're', 'string', 'sys', 'time', 'traceback', 'uuid', 'os', 'base64',
-    'hashlib', 'hmac', 'binascii', 'collections', 'functools', 'operator', 'itertools', 'uuid',
+    'hashlib', 'hmac', 'binascii', 'collections', 'functools', 'operator', 'itertools',
```
Comment (Yeuoly) [inline api/core/helper/code_executor/python3/python3_transformer.py]: Is this an irrelevant change?

Comment (bowenliang123) [reply api/core/helper/code_executor/python3/python3_transformer.py]: No, this is auto-fixed by the rule B033 `duplicate-value` (Checks for set literals that contain duplicate items.).
`uuid` is duplicated in PYTHON_STANDARD_PACKAGES on line 35

Comment (Yeuoly) [reply api/core/helper/code_executor/python3/python3_transformer.py]: Okay, I saw it, thx.

**Review by Yeuoly — DISMISSED** [review]
LGTM

**Comment by Yeuoly** [comment]
hello, there is a conflict need to be solved.

**Comment by bowenliang123** [comment]
The conflict has been resolved.

**Review by Yeuoly — APPROVED** [review]
LGTM

**Pull request merged**